docs/reference/module-overview.md

```diff
@@ -0,0 +1,284 @@
+<!-- TODO: link to a definition of a bundle when available. -->
+
+You might include this in a stand-alone module that adds functionality to the `@apostrophecms/image` core module:
+
+```javascript
+// index.js
+modules.export = {
+  improve: '@apostrophecms/image'
+  // Additional functionality ...
+};
+```
+
+You should not use both `improve` and `extend` in a single module.
+
+
+### `options`
+
+An object used to add additional, often optional, settings to a module. There are many options available depending on the module type. See the [module configuration options page](/reference/module-options.md) for more information.
+
+### `fields`
+
+[Doc type](/reference/glossary.md#doc) modules have some fields configured by default, such as the `title` and `slug` fields. The `fields` setting is used for additional field management.
+
+The `fields` object is configured with subsections: `add`, `remove`, and `group`.
+
+#### `add`
+
+An object of fields to add to the doc type's schema. See the [field type reference](/reference/field-types/README.md) for more on field type configuration.
+
+```javascript
+// modules/article/index.js
+modules.export = {
+  fields: {
+    add: {
+      subtitle: {
+        label: 'Subtitle',
+        type: 'string'
+      }
+    }
+  }
+};
+```
+
+#### `remove`
+
+An array of field names from the base class module to remove. Some core doc type fields cannot be removed since they required by core functionality.
+
+```javascript
+// modules/spotlight-article/index.js
+modules.export = {
+  extend: 'article',
+  fields: {
+    remove: [ 'subtitle' ]
+  }
+};
+```
+
+#### `group`
+
+An object of field groups. The groupings are used by editing interface. It _does not apply to widget modules_, which do not have a tabbed interface.
+
+Groups are added as an object with their name as the object key and the following properties:
+- `label`: The visible label (a string) on the interface tab
+- `fields`: An array of field names to include in the group
+
+The `@apostrophecms/doc-type` module groups the default fields in a `basics` group, which you can override. If overriding the `basics` group, you will need to add those core fields to a group in the module. Any fields not added to a group will be placed in an "Ungrouped" section in the editing interface.
+
+```javascript
+// modules/article/index.js
+modules.export = {
+  fields: {
+    add: {
+      // ...
+    },
+    group: {
+      meta: { // 👈 The group's identifying name is the object key.
+        label: 'Article metadata',
+        fields: [
+          'subtitle',
+          'author',
+          '_category'
+        ]
+      }
+    }
+  }
+};
+```
+
+### `filters`
+
+In piece type modules, the `filters` setting configures the pieces manager modal, adding and removing filtering interface (to view only certain pieces). `trash` and `visibility` filters are included by default.
```

Comment:
"the pieces manager modal" -- I think this is the first time in the docs ive seen this referenced. 

1. we might want to follow up with a definition or link at some point
2. Lets not call it a modal, its not relevant to describe its design, just its function.

",adding and removing filtering interface (to view only certain pieces)" -- this doesn't feel like a complete sentence.